WalletWasabi/Crypto/StringCipher.cs

```diff
@@ -1,114 +1,115 @@
-using System.IO;
 using System.Linq;
 using System.Security.Cryptography;
 using System.Text;
 using WalletWasabi.Crypto.Randomness;
 
 namespace WalletWasabi.Crypto
 {
-	// https://stackoverflow.com/a/10177020/2061103
 	public static class StringCipher
 	{
 		// This constant is used to determine the keysize of the encryption algorithm in bits.
 		// We divide this by 8 within the code below to get the equivalent number of bytes.
-		private const int KeySize = 128;
+		private const int KeyByteSize = 128 / 8;
+		private const int AesBlockByteSize = 128 / 8;
+		private const int PasswordSaltByteSize = 128 / 8;
+		private const int SignatureByteSize = 256 / 8;
 
 		// This constant determines the number of iterations for the password bytes generation function.
 		private const int DerivationIterations = 1000;
 
+		private const int MinimumEncryptedMessageByteSize =
+			PasswordSaltByteSize + // Auth salt
+			PasswordSaltByteSize + // Key salt
+			AesBlockByteSize + // IV
+			AesBlockByteSize + // Cipher text min length
+			SignatureByteSize; // Signature tag
+
+		private static readonly SecureRandom SecureRandom = new();
+
 		public static string Encrypt(string plainText, string passPhrase)
 		{
 			// Salt is randomly generated each time, but is prepended to encrypted cipher text
 			// so that the same Salt value can be used when decrypting.
-			byte[] salt = Generate128BitsOfRandomEntropy();
-			byte[] cipherTextBytes;
-			var plainTextBytes = Encoding.UTF8.GetBytes(plainText);
-
-			byte[] key = DerivateKey(passPhrase, salt);
-			using var aes = CreateAES();
-			aes.GenerateIV();
-			byte[] iv = aes.IV;
-			using var encryptor = aes.CreateEncryptor(key, iv);
-			using var memoryStreamEncryptor = new MemoryStream();
-			using (var cryptoStream = new CryptoStream(memoryStreamEncryptor, encryptor, CryptoStreamMode.Write))
-			{
-				cryptoStream.Write(plainTextBytes, 0, plainTextBytes.Length);
-				cryptoStream.FlushFinalBlock();
-			}
-			cipherTextBytes = memoryStreamEncryptor.ToArray();
-
-			using var memoryStream = new MemoryStream();
-			using var writer = new BinaryWriter(memoryStream);
-			writer.Write(salt);
-			writer.Write(iv);
-			using var hmac = new HMACSHA256(key);
-			var authenticationCode = hmac.ComputeHash(iv.Concat(cipherTextBytes).ToArray());
-			writer.Write(authenticationCode);
-			writer.Write(cipherTextBytes);
-			writer.Flush();
-
-			var cipherTextWithAuthBytes = memoryStream.ToArray();
-			return Convert.ToBase64String(cipherTextWithAuthBytes);
+			byte[] iv = GenerateRandomEntropy(AesBlockByteSize);
+			byte[] encryptionKeySalt = GenerateRandomEntropy(PasswordSaltByteSize);
+			byte[] encryptionKey = DerivateKey(passPhrase, encryptionKeySalt);
+
+			// Encrpyt the plain text
+			using var aes = CreateAES(encryptionKey);
+			byte[] plainTextBytes = Encoding.UTF8.GetBytes(plainText);
+			byte[] cipherTextBytes = aes.EncryptCbc(plainTextBytes, iv);
```

Comment:
Maybe it's worth specifying even the padding mode explicitly (+ also in Decrypt):

```suggestion
			byte[] cipherTextBytes = aes.EncryptCbc(plainTextBytes, iv, PaddingMode.PKCS7);
```

if we want to be sure that there won't be a breaking change with a new .NET release (I would guess it's unlikely though.)

Reply:
it is the default, i think it doesn't add anything.